app/src/main/java/org/fossasia/openevent/general/ProfileFragment.java

```diff
@@ -86,31 +76,40 @@ public void onClick(View v) {
 
         progressBar.setIndeterminate(true);
-
-        compositeDisposable.add(ApiClient.getClient2(TOKEN).getProfile(userId)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(response -> {
-                    if(response.isSuccessful()){
-
-                        progressBar.setIndeterminate(false);
-                        progressBar.setVisibility(View.GONE);
-                        Timber.d("Response Success");
-                        User userAttrib = response.body();
-                        firstNameTv.setText(userAttrib.getFirstName());
-                        emailTv.setText(userAttrib.getEmail());
-
-                        Picasso.with(view.getContext())
-                                .load(userAttrib.getAvatarUrl())
-                                .placeholder(R.drawable.ic_person_black_24dp)
-                                .transform(new CircleTransform())
-                                .into(avatarImageView);
-                    } else {
-                        Timber.d("Not Successfull, Error code : "+response.code());
-                    }
-                }, throwable -> {
-                    Timber.e("Failure" + "\n" + throwable.toString());
-                }));
+        int id=-1;
+        try {
+            id = JWTUtils.getIdentity(AuthUtil.getAuthorization());
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        if(id!=-1) {
+            compositeDisposable.add(ApiClient.getOpenEventAPI().getProfile(id)
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(response -> {
+                        if (response.isSuccessful()) {
+
+                            progressBar.setIndeterminate(false);
+                            progressBar.setVisibility(View.GONE);
+                            Timber.d("Response Success");
+                            User userAttrib = response.body();
+                            firstNameTv.setText(userAttrib.getFirstName());
+                            emailTv.setText(userAttrib.getEmail());
+
+                            Picasso.with(view.getContext())
+                                    .load(userAttrib.getAvatarUrl())
+                                    .placeholder(R.drawable.ic_person_black_24dp)
+                                    .transform(new CircleTransform())
+                                    .into(avatarImageView);
+                        } else {
+                            Timber.d("Not Successfull, Error code : " + response.code());
+                        }
+                    }, throwable -> {
+                        Timber.e("Failure" + "\n" + throwable.toString());
+                    }));
+        } else {
+            Toast.makeText(getContext(),"Try later",Toast.LENGTH_SHORT).show();
+        }
```

Comment:
Doing too much here

Reply:
this will be moved to view model in the coming PR's . let keep it this way for now. Opening issue for moving logics into view model

Reply:
Still this can be broken into functions

Reply:
okay, will change this